zerver/views/realm_export.py

```diff
@@ -97,20 +96,17 @@ def get_realm_exports(request: HttpRequest, user: UserProfile) -> HttpResponse:
 @require_realm_admin
 def delete_realm_export(request: HttpRequest, user: UserProfile, export_id: int) -> HttpResponse:
     try:
-        audit_log_entry = RealmAuditLog.objects.get(
-            id=export_id, realm=user.realm, event_type=AuditLogEventType.REALM_EXPORTED
-        )
-    except RealmAuditLog.DoesNotExist:
+        export_row = RealmExport.objects.get(id=export_id)
+    except RealmExport.DoesNotExist:
         raise JsonableError(_("Invalid data export ID"))
 
-    export_data = audit_log_entry.extra_data
-    if export_data.get("deleted_timestamp") is not None:
+    if export_row.status == RealmExport.DELETED:
         raise JsonableError(_("Export already deleted"))
-    if export_data.get("export_path") is None:
-        if export_data.get("failed_timestamp") is not None:
-            raise JsonableError(_("Export failed, nothing to delete"))
+    if export_row.status == RealmExport.FAILED:
+        raise JsonableError(_("Export failed, nothing to delete"))
+    if export_row.status in [RealmExport.REQUESTED, RealmExport.STARTED]:
         raise JsonableError(_("Export still in progress"))
-    do_delete_realm_export(audit_log_entry)
+    do_delete_realm_export(export_row)
     return json_success(request)
```

Comment:
We probably will want to add a new `RealmAuditLog` type for deleting an export, such that `acting_user` can be tracked properly. This can be a quick follow-up.